modules/_labs/side-panel/react/lib/SidePanel.tsx

```diff
@@ -0,0 +1,256 @@
+
+export type SidePanelVariant = 'standard' | 'alternate';
+export type SidePanelTransitionStates = 'collapsed' | 'collapsing' | 'expanded' | 'expanding';
+
+export interface SidePanelProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The element the side panel will render as (e.g. 'div').
+   *
+   * @default 'section'
+   */
+  as?: React.ElementType;
+  /**
+   * The width of the component (in `px` if it's a `number`) when it is collapsed.
+   *
+   * @default 64
+   */
+  collapsedWidth?: number | string;
+  /**
+   * If true, sets the expanded state of the side panel
+   * @default true
+   */
+  expanded?: boolean;
+  /**
+   * The width of the component (in `px` if it's a `number`) when it is expanded.
+   *
+   * @default 320
+   */
+  expandedWidth?: number | string;
+  /**
+   * The height of the component (in `px` if it's a `number`).
+   *
+   * @default '100%'
+   */
+  height?: number | string;
+  /**
+   * Which side the side panel is meant to originate from.
+   *
+   * @default 'left'
+   */
+  origin?: 'left' | 'right';
+  /**
+   * The function called when the side panel's `expanded` state changes. States like `'collapsing'` and `'expanding'` are tracked in another callback: `onStateChange`
+   *
+   * @param boolean
+   */
+  onExpandedChange?: (expanded?: boolean) => void;
+  /**
+   * The function called when the side panel is transitioning between states.
+   * Use this to track when the side panel is animating between `'collapsed'`, `'collapsing'`, `'expanded'`, and `'expanding'` states.
+   * This can be particularly helpful if child components need to react specifically to these states.
+   *
+   * @param SidePanelTransitionStates
+   */
+  onStateTransition?: (state?: SidePanelTransitionStates) => void;
+  /**
+   * The style variant of the side panel. 'standard' is with a `soap100` background, no depth. 'alternate' is a `frenchVanilla100` background with a level 3 depth.
+   *
+   * @default 'standard'
+   */
+  variant?: SidePanelVariant;
+}
+
+const createKeyframes = (from: number | string, to: number | string) => {
+  const normalized = {
+    from: typeof from === 'number' ? from + 'px' : from,
+    to: typeof to === 'number' ? to + 'px' : to,
+  };
+
+  return keyframes`
+    from {
+      width: ${normalized.from};
+      min-width: ${normalized.from};
+      max-width: ${normalized.from};
+    } to {
+      width: ${normalized.to};
+      min-width: ${normalized.to};
+      max-width: ${normalized.to};
+    }
+  `;
+};
+
+const containerVariantStyle: {[K in SidePanelVariant]: CSSObject} = {
+  alternate: {
+    backgroundColor: colors.frenchVanilla100,
+    ...depth[3],
+  },
+  standard: {
+    backgroundColor: colors.soap100,
+  },
+};
+
+const Panel = styled('section')<Pick<SidePanelProps, 'as'>>({
+  overflow: 'hidden',
+  position: 'relative',
+  boxSizing: 'border-box',
+});
+
+export const SidePanelContext = React.createContext({
+  state: 'expanded',
+  origin: 'left',
+});
```

Comment:
Devil's advocate: 👿 Since this context is only being used in the `ToggleButton`, and `ToggleButton` is already taking `{...controlProps}`, could this just live in the hook props? Is context more helpful here?